tests/client/check_client_subscriptions.c

```diff
@@ -1167,6 +1228,14 @@ stateCallback(UA_Client *client, UA_SecureChannelState channelState,
         /* A new session was created. We need to create the subscription. */
         UA_CreateSubscriptionRequest request = UA_CreateSubscriptionRequest_default();
         request.requestedMaxKeepAliveCount = 1;
+        /*
+         * TODO: creating synchronious subscription in stateCallback is illegal --
+         * The stateCallback is called when the eventloop is within the running and
+         * a change of the client state is notifier. The synchronous subscription call
+         * below needs to iterate the eventloop another time because it needs to get
+         * the response for the subscription request.
+         * This is illegal How to deal with this?
+         * /
         UA_CreateSubscriptionResponse response = UA_Client_Subscriptions_create(client, request,
```

Comment:
When the configured stateCallback is executed the client will be within an eventloop-iteration, as the for example the stateCallback is called in response to a received message from the current connection with a server.

`UA_Client_Subscriptions_create` in turn is a synchronous service execution on the server by a client. This means, a request will be send and a corresponding response will be awaited. This means the eventloop-needs to run until the corresponding response is received on that connection. Running the eventloop in such a recursive manner is illegal. How to deal with this?

Possibilities:

1. Render such definition of callbacks illegal. Might introduce a braking change for user-code that needed this behaviour
2. Investigate the possibility to use asynchronous service calls in this case (might also introduce braking changes as the response is expected to be returned by this service call)
3. Make the eventloop robust against recursive calls and allow them
4. Always run the eventloop in a dedicated thread (This would require true multithreading by the OS, so actually this is not an option)
5. ...?